src/projects/shared/site/src/containers/ImageExporterPopup/index.tsx

```diff
@@ -16,33 +16,40 @@ import {CircuitDesigner} from "shared/api/circuitdesigner/public/CircuitDesigner
 import {DefaultTool}                     from "shared/api/circuitdesigner/tools/DefaultTool";
 import {PanTool}                         from "shared/api/circuitdesigner/tools/PanTool";
 import {FitToScreen, FitToScreenHandler} from "shared/api/circuitdesigner/tools/handlers/FitToScreenHandler";
+import {ZoomHandler}                     from "shared/api/circuitdesigner/tools/handlers/ZoomHandler";
 
 import {ButtonToggle} from "shared/site/components/ButtonToggle";
 import {InputField}   from "shared/site/components/InputField";
 import {Popup}        from "shared/site/components/Popup";
 import {SwitchToggle} from "shared/site/components/SwitchToggle";
 
 import "./index.scss";
+import {useWindowSize} from "shared/site/utils/hooks/useWindowSize";
+import {CircuitHelpers} from "shared/site/utils/CircuitHelpers";
+import {ToolHandler} from "shared/api/circuitdesigner/tools/handlers/ToolHandler";
 
 
 const MIN_IMG_SIZE = 50;
 const MAX_IMG_SIZE = 10_000;
 
 type Props = {
     designer: CircuitDesigner;
+    extraHandlers: ToolHandler[];
 }
-export const ImageExporterPopup = ({ designer }: Props) => {
+export const ImageExporterPopup = ({ designer, extraHandlers }: Props) => {
     const { curPopup, circuitName } = useSharedSelector(
         (state) => ({ curPopup: state.header.curPopup, circuitName: state.circuit.name })
     );
     const dispatch = useSharedDispatch();
 
     const isActive = (curPopup === "image_exporter");
 
+    const { w, h } = useWindowSize();
```

Comment:
If you're using this then you don't need the whole useEffect down below subscribing to resize; and width and height don't need to be state

Reply:
The state width/height can be changed by these options in the popup though 
![image](https://github.com/user-attachments/assets/73772dcb-4137-4c85-8b60-b72f8ba13718)

Reply:
Just realized that w/h are only used to set the initial state though, so I'll just remove that usage and directly use window.innerWidth/Height instead